validator.go

```diff
@@ -109,3 +108,19 @@ func newValidatorImpl(domains []string, usersFile string,
 func NewValidator(domains []string, usersFile string) func(string) bool {
 	return newValidatorImpl(domains, usersFile, nil, func() {})
 }
+
+// Check if the authenticated email is validated against the provided domain
+func isEmailValidWithDomains(email string, emailDomain string) bool {
```

Comment:
We should take in `emailDomains []string` instead of `emailDomain string` here and manage the looping in this function.

Then we don't need this logic above:

```
		for _, domain := range domains {
			valid = valid || isEmailValidWithDomains(email, domain)
		}
```

Reply:
Done.